sdk/trace/batch_span_processor.go

```diff
@@ -167,33 +174,34 @@ func WithBlocking() BatchSpanProcessorOption {
 	}
 }
 
-// processQueue removes spans from the `queue` channel until there is
-// no more data.  It calls the exporter in batches of up to
-// MaxExportBatchSize until all the available data have been processed.
+// processQueue removes spans from the `queue` channel until the queue
+// fills once or there is no more data.  It calls the exporter once with
+// a batch of up to MaxExportBatchSize.  It sends the `eof` boolean to
+// true when a `nil` *SpanData is received, indicating that the processor
+// is flushing.
 func (bsp *BatchSpanProcessor) processQueue(batch *[]*export.SpanData) {
-	for {
```

Comment:
Removing this loop means that the processor will only process one batch per tick, right? Is that desired?

The original outer `for` loop allowed for more than the max batch size number of spans to be exported in a single tick, just in batch up to the max match size.

Reply:
This is the (intentional) sense in which this PR undoes #599. 
The original poster (@vmihailenco) in #702 was against the behavior of that outer `for` loop because in a steady state, there will always be more spans arriving in small batches.

I had been assuming that the output of the batch span processor would be an RPC, and that it is a good thing to eagerly send a batch whenever the RPC sender is not busy. Thus, you'd keep sending as fast as possible as long as there are spans. This naturally allows the batch to grow as large as it must during one RPC, which I think of as relatively good/simple behavior. IOW the former approach only gives you small batches when the processor is very fast, which is the case for @vmihailenco.

Reply:
Ah, gotcha, that does look correct. These changes do seem to address the case where you have a small number of spans the you want to build up into a larger batch. However, it seems to be doing this in the queue, not the batching. It assumes that there will only be one export per call to `processQueue`.

This assumption is correct when you want to batch many small addition of spans being added to the queue, but it will fail in some pathological cases (i.e. small enqueues followed by very large enqueues, or enqueues made at a rate greater than `bsp.o.MaxExportBatchSize/bsp.o.ScheduledDelayMillis`).

It seems like the important thing from #702 is the selecting across the queue or ticker (or stop chan, but not as relevant here). This means that if the buffer fills in an interval `bsp.o.ScheduledDelayMillis` it will be flushed and another will buffer started, otherwise it will flush the buffer at most once per `bsp.o.ScheduledDelayMillis`.

This approach only supports the latter, flushing once per `bsp.o.ScheduledDelayMillis`.

Reply:
I got into this mess because of a flaky test. At least we now understand the change in #702.

The comment above `ScheduledDelayMillis` says 

> ScheduledDelayMillis is the delay interval in milliseconds between two consecutive processing of batches.

which makes the behavior of this PR sound "right". If we go back to #702, all of this explanation is what I was requesting, and now we have answers.